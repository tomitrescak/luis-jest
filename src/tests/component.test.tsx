import * as React from 'react';

storyOf('Input', () => <div>My Component</div>);

// you can group tests using BDD approach, with each describe generating one folder in
describe('Folder', function() {
  storyOf(
    'Component With Test',
    {
      foo: 2,
      bar: 3,
      get component() {
        // just another notation
        return <div>My Tested</div>;
      }
    },
    data => {
      it('passes', () => {
        expect(data.foo).toEqual(2);
      });

      it('fails', () => {
        expect(data.bar).toEqual(3);
      });
    }
  );

  storyOf(
    'Other Component',
    {
      get component() {
        return (
          <div>Other Component 7</div>
        );
      }
    },
    data => {
      it('renders correctly', function() {
        expect(data.component).toMatchSnapshot();
      });

      it('renders other', function() {
        expect(data.component).toMatchSnapshot();
      });
    }
  );
});
