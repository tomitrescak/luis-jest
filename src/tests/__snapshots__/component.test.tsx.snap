// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folder Other Component renders correctly 1`] = `
<div>
  Other Component 7
</div>
`;

exports[`Folder Other Component renders other 1`] = `
<div>
  Other Component 7
</div>
`;
